src/main/java/com/mlaf/hu/decisionagent/DecisionAgent.java

```diff
@@ -0,0 +1,102 @@
+package com.mlaf.hu.decisionagent;
+
+import com.mlaf.hu.decisionagent.behavior.ReceiveBehavior;
+import com.mlaf.hu.decisionagent.behavior.RegisterBehavior;
+import com.mlaf.hu.decisionagent.behavior.UpdateBehavior;
+import com.mlaf.hu.helpers.JadeServices;
+import com.mlaf.hu.models.*;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.domain.DFService;
+import jade.util.Logger;
+
+import javax.xml.bind.JAXB;
+import java.io.StringReader;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public abstract class DecisionAgent extends Agent {
+    private static final String SERVICE_NAME = "DECISION-AGENT";
+    private static final int MAX_READINGS = 100;
+    public static java.util.logging.Logger decisionAgentLogger = Logger.getLogger("DecisionAgentLogger");
+    public HashMap<AID, InstructionSet> sensorAgents = new HashMap<>();
+
+    public DecisionAgent() {
+        super();
+    }
+
+    @Override
+    protected void setup() {
+        try {
+            JadeServices.registerAsService(SERVICE_NAME, "decision-agent", null, null, this);
+            addBehaviour(new RegisterBehavior(this));
+            addBehaviour(new ReceiveBehavior(this));
+            addBehaviour(new UpdateBehavior(this));
+        } catch (Exception e) {
+            DecisionAgent.decisionAgentLogger.log(Logger.SEVERE, "Could not initialize BrokerAgent", e);
+            System.exit(1);
+        }
+    }
+
+    protected void takeDown() {
+        try {
+            DFService.deregister(this);
+        } catch (Exception ignore) {
+        }
+    }
+
+    public void registerSensorAgent(AID sensoragent, InstructionSet instructionset) {
+        this.sensorAgents.put(sensoragent, instructionset);
+    }
+
+    public abstract void unregisterSensorAgent(AID sensoragent);
+
+    public InstructionSet parseInstructionXml(String xml) {
+        InstructionSet is = new InstructionSet();
+        try {
+            is = JAXB.unmarshal(new StringReader(xml), InstructionSet.class);
+        } catch (Exception e) {
+            decisionAgentLogger.log(Logger.SEVERE, String.format("Error parsing XML: %s", e.getMessage()));
```

Comment:
Ik denk dat als een instructionset niet gelezen kan werden we een groter probleem hebben dan we met een "log kunnen handlen.
Misschien eigen exception?

Reply:
Het raisen van exceptions knalt de agent eruit als ik me niet vergis. Is dat wat we willen of laten we de programmeurs de Sensor Agents' instructieset aanpassen en het nog een keer proberen? Als het goed is heeft de InstructieSet wel een antwoord klaar voor als er iets niet goed is aan de opmaak.

Reply:
Ik denk dat we die exception weer op een hoger niveau moeten afvangen en een NOTUNDERSTOOD of hoe die performative ook heet terug moeten sturen